src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java

```diff
@@ -782,18 +749,16 @@ public ArtifactVersions lookupDependencyUpdates( Dependency dependency, boolean
         throws ArtifactMetadataRetrievalException, InvalidVersionSpecificationException
     {
         // Create the request for details collection for parallel lookup...
-        final List<Callable<PluginPluginUpdatesDetails>> requestsForDetails =
-            new ArrayList<Callable<PluginPluginUpdatesDetails>>( plugins.size() );
+        List<Callable<PluginPluginUpdatesDetails>> requestsForDetails = new ArrayList<>( plugins.size() );
         for ( final Plugin plugin : plugins )
         {
             requestsForDetails.add( new PluginLookup( plugin, allowSnapshots ) );
         }
 
-        final Map<Plugin, PluginUpdatesDetails> pluginUpdates =
-            new TreeMap<Plugin, PluginUpdatesDetails>( new PluginComparator() );
+        Map<Plugin, PluginUpdatesDetails> pluginUpdates = new TreeMap<>( new PluginComparator() );
 
         // Lookup details in parallel...
-        final ExecutorService executor = Executors.newFixedThreadPool( LOOKUP_PARALLEL_THREADS );
+        ExecutorService executor = Executors.newFixedThreadPool( LOOKUP_PARALLEL_THREADS );
```

Comment:
Why remove the final?

Reply:
because this has not value here (context is a local variable available for only 15 lines)
except noise this final worths nothing

Reply:
and I would remove even more useless final